pycytominer/cyto_utils/cells.py

```diff
@@ -410,9 +438,30 @@ def load_compartment(self, compartment):
         pandas.core.frame.DataFrame
             Compartment dataframe.
         """
-        compartment_query = "select * from {}".format(compartment)
-        df = pd.read_sql(sql=compartment_query, con=self.conn)
-        return df
+
+        # Get data useful to pre-alloc memory
+        num_cells = self.count(compartment)
+        meta_cols, feat_cols = self.get_columns(compartment)
+        num_meta, num_feats = len(meta_cols), len(feat_cols)
+
+        # Use pre-allocated np.array for data
+        feats = np.empty(shape=(num_cells, num_feats), dtype=np.float64)
+        # Use pre-allocated pd.DataFrame for metadata
+        metas = pd.DataFrame(columns=meta_cols, index=range(num_cells))
+
+        # Query database for selected columns of chosen compartment
+        columns = ", ".join(meta_cols + feat_cols)
+        query = f"select {columns} from {compartment}"
+        resultset = self.conn.execute(query)
+
+        # Load data row by row for both meta information and features
+        for i, row in enumerate(resultset):
+            metas.loc[i] = row[:num_meta]
+            feats[i] = row[num_meta:]
```

Comment:
Is there any reason to not include the row _and_ column indexing here?

(e.g. `metas.loc[i, :] = row[:num_meta=]`, and something equivalent for `feats`)

To me, it's easier to understand that the variables you're building are 2d if we explicitly set both indices. 

We shouldn't implement my suggestion if it slows processing